Index: b/usbmount
===================================================================
--- a/usbmount
+++ b/usbmount
@@ -91,6 +91,7 @@
     FSTYPE=$(echo "$DEVINFO" | sed 's/.*[[:blank:]]TYPE="\([^"]*\)".*/\1/g; s/[[:blank:]]*//g;')
     UUID=$(echo "$DEVINFO"   | sed 's/.*[[:blank:]]UUID="\([^"]*\)".*/\1/g; s/[[:blank:]]*//g;')
     USAGE=$(echo "$DEVINFO"  | sed 's/.*[[:blank:]]USAGE="\([^"]*\)".*/\1/g; s/[[:blank:]]*//g;')
+    LABEL=$(echo "$DEVINFO"  | sed 's/.*[[:blank:]]LABEL="\([^"]*\)".*/\1/g; s/[[:blank:]]*//g;')
 
     if ! echo $USAGE | egrep -q "(filesystem|disklabel)"; then
 	log info "$DEVNAME does not contain a filesystem or disklabel"
@@ -121,6 +122,10 @@
 		    break
 		fi
 	    done
+            if [ $LABEL == 'USERDATA' ] ; then
+                mountpoint=""
+                log info "mount of $DEVNAME aborted due to LABEL == USERDATA"
+            fi
 	    if [ -n "$mountpoint" ]; then
 		# Determine mount options.
 		options=
